AirCasting/SearchAndFollow/SearchLocation/SearchViewModel.swift

```diff
@@ -2,13 +2,32 @@
 //
 
 import Foundation
+import CoreLocation
+import Resolver
 
 class SearchViewModel: ObservableObject {
+    @Injected private var networkChecker: NetworkChecker
     
     @Published var isLocationPopupPresented = false
-    @Published var location = ""
+    @Published var addressName = ""
+    @Published var addresslocation = CLLocationCoordinate2D(latitude: 20.0, longitude: 20.0)
+    @Published var alert: AlertInfo?
     
     func textFieldTapped() { isLocationPopupPresented.toggle() }
     
-    func updateLocation(using newLocation: String) { location = newLocation}
+    func updateLocationName(using newLocationName: String) {
```

Comment:
this is not how a ViewModel API should be. It should have inputs related to UI actions, not application logic actions. So for example a viewmodel handling a button that logs the user out should not have a function named `logoutUser()`, rather it should be `logoutButtonTapped()`. View doesn't tell viewmodel **what to do**, it tells it **what happened**